misk-redis/src/main/kotlin/misk/redis/JedisPooledWithMetrics.kt

```diff
@@ -34,7 +34,7 @@ private class ConnectionFactoryWithMetrics(
   requiresPassword: Boolean = true,
 ) : ConnectionFactory(
   HostAndPort(
-    replicationGroupConfig.writer_endpoint.hostname,
+    replicationGroupConfig.writer_endpoint.hostname?.takeUnless { it.isNullOrBlank() } ?: System.getenv("REDIS_HOST") ?: "127.0.0.1",
```

Comment:
Misk doesn't even configure the reader endpoint for RedisConfig, so its just the writer that is being updated.